be2-scala/src/main/scala/ch/epfl/pop/model/objects/LaoData.scala

```diff
@@ -9,9 +9,15 @@ import ch.epfl.pop.model.network.method.message.data.lao.CreateLao
 import ch.epfl.pop.model.network.method.message.data.rollCall.CloseRollCall
 import spray.json._
 
+//Ed25519 is used to verify the signatures, therefore I also use it to sign data
+import com.google.crypto.tink.subtle.Ed25519Sign
+
+//a pair of byte arrays works better as keypair than the actual keypair object, since Ed25519Sign.Keypair isn't directly convertible to json with spray
 case class LaoData(
     owner: PublicKey,
     attendees: List[PublicKey],
+    privateKey: Array[Byte], 
+    publicKey: Array[Byte],
```

Comment:
Mhmmm I would still use the objects we define here. Otherwise, we use all the benefits of OOP for keys. You can use PublicKey and a new PrivateKey objects here. PrivateKey does not exist (yet) because we had no reason (up until now) to treat private keys in the server. You could then add the function `signData` you defined in SocialMediaHandler as a method of private key

And finally, to address the comment `//a pair of byte arrays works better as keypair than the actual keypair object, since Ed25519Sign.Keypair isn't directly convertible to json with spray`. If, at keypair creation, you convert them to a PublicKey and PrivateKey, you'll have no issue. The reason behind it is that we tell spray-json how to encode/decode PublicKeys (see json/ObjectProtocol.scala:35):
```scala
  implicit object PublicKeyFormat extends JsonFormat[PublicKey] {
    override def read(json: JsValue): PublicKey = json match {
      case dataJs@JsString(_) => PublicKey(dataJs.convertTo[Base64Data])
      case _ => throw new IllegalArgumentException(s"Can't parse json value $json to a PublicKey object")
    }

    override def write(obj: PublicKey): JsValue = obj.base64Data.toJson
  }
```

Of course you'd have to copy-paste this with minor adjustements for a PrivateKey